pkg/multiTenancyPlugins/naming/nameScoping.go

```diff
@@ -248,3 +272,24 @@ func getContainerId(container *cluster.Container, tenantId string, containerRefe
 	}
 	return "", errors.New(CONTAINER_REFERENCE_NOT_FOR_CONTAINER_INFO)
 }
+
+// send request to next handler in record mode.  When it returns remove traces of tenant id in response
+func doReq(nameScoping *DefaultNameScopingImpl, command utils.CommandEnum, cluster cluster.Cluster, w http.ResponseWriter, r *http.Request, swarmHandler http.Handler) error {
```

Comment:
Why not just do the manipulation in the function and create a recorder and send to nect handler from the main case statement.